source/includes/k8s/file-transfer-protocol-k8s.rst

```diff
@@ -0,0 +1,167 @@
+
+
+Considerations
+--------------
+
+
+Versioning
+~~~~~~~~~~
+
+SFTP clients cannot read specific :ref:`object versions <minio-bucket-versioning>` other than the latest version.
+
+- For read operations, MinIO only returns the latest version of the requested object(s) to the sftp client.
+- For write operations, MinIO applies normal versioning behavior for matching object names.
+
+Use an S3 API Client, such as the :ref:`MinIO Client <minio-client>`.
+
+
+Authentication and Access
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``SFTP`` access requires the same authentication as any other S3 client.
+MinIO supports the following authentication providers:
+
+- :ref:`MinIO IDP <minio-internal-idp>` users and their service accounts
+- :ref:`Active Directory/LDAP <minio-external-identity-management-ad-ldap>` users and their service accounts
+- :ref:`OpenID/OIDC <minio-external-identity-management-openid>` service accounts
+
+:ref:`STS <minio-security-token-service>` credentials **cannot** access buckets or objects over SFTP.
+To use STS credentials to authenticate, you must use an S3 API client or port.
+
+Authenticated users can access buckets and objects based on the :ref:`policies <minio-policy>` assigned to the user or parent user account.
+
+The SFTP protocol does not require any of the ``admin:*`` :ref:`permissions <minio-policy-mc-admin-actions>`.
+You may not perform other MinIO admin actions with SFTP.
+
+
+Prerequisites
+-------------
+
+- MinIO Operator v5.0.7 or later.
+- Enable an SFTP port (8022) for the server.
+- A port to use for the SFTP commands and a range of ports to allow the SFTP server to request to use for the data transfer.
+
+
+Procedure
+---------
+
+#. Enable SFTP for the desired Tenant:
+
+   .. tab-set::
+
+      .. tab-item:: Operator Console
+
+         - In the Operator Console, click on the tenant for which to enable SFTP.
+         - In the :guilabel:`Configuration` tab, toggle :guilabel:`SFTP` to :guilabel:`Enabled`.
+         - Click :guilabel:`Save`.
+         - Click :guilabel:`Restart` to restart MinIO and apply your changes.
+
+      .. tab-item:: Kubectl
+
+	 Use the following command to edit the Tenant YAML configuration:
+
+	 .. code-block:: yaml
+			 
+            kubectl edit tenants/my-tenant -n my-tenant-ns
+
+	 Replace ``my-tenant`` and ``my-tenant-ns`` with the desired Tenant and namespace.
+
+         In the ``features:`` section, set the value of ``enableSFTP`` to ``true``:
+
+         .. code-block:: yaml
+
+            spec:
+              configuration:
+                name: my-tenant-env-configuration
+              credsSecret:
+                name: my-tenant-secret
+              exposeServices:
+                console: true
+                minio: true
+              features:
+                enableSFTP: true
+
+         Kubectl restarts MinIO to apply the change.
+
+         You may also set ``enableSFTP`` in your Helm chart or Kustomize configuration to enable SFTP for newly created Tenants.
```

Comment:
Maybe link to the examples in the Operator repo? They are reasonably well commented/easy to follow.

This is skirting rather close to explaining Helm & Kustomize syntax, but without noting this you might easily miss that you can do this.

Reply:
@feorlen what examples are you referring to?, these: https://github.com/minio/operator/tree/master/examples/kustomization
If so then maybe yes, but I don't see any in particullar for sftp, would be nice to have one for testing!.

Reply:
@cniackz Yes, this one specifically that shows the new `enableSFTP` setting:  https://github.com/minio/operator/blob/8385948929bc95648d1be82d96f829c810519674/examples/kustomization/base/tenant.yaml#L28

Eventually we will cover the contents of these YAML files (Helm/Kustomize) but for the moment we presume anyone needing those knows what to expect when looking through them. 

Although since FTP/SFTP turns out to be a popular feature, worth a brief mention. After all, we recommend Kubernetes users go with Operator.

Reply:
I linked to relevant examples, although to the whole file rather than a line number that will rapidly be out of date. Not my favorite solution, but a sufficient pointer for folks who already use Helm or Kustomize.